### DISCUSSION TITLE
Is there an expiration time setting for user sessions in Atoti+?

### DISCUSSION THREAD

**Discussion by tht-activeviam** [atoti/atoti#533] [Q&A]
How do we set the timeout of Atoti session?
I couldn't find in session_config. Do we have any default timeout value at the moment?

**Comment by bloussou** [comment]
Hello,
Are you talking about queries timeout?
If yes, it is using [`Cube.shared_context`](https://docs.atoti.io/latest/lib/atoti.cube.html#atoti.cube.Cube.shared_context) with a default to *30s*. For each query, you can choose a specific timeout by setting the *timeout* parameter.

**Reply by tht-activeviam** [reply]
Hi,
I am referring to user session timeout. We are running Atoti as standalone, not from the python Notebook.
For production, we are going through the authentication test cases.
- When the user idle for a certain time, they should be logged out and redirect to login page. Do we have that timeout setting to expire the session?
- When the user session is still active but the login JWT token is expired, did we log out the user and redirect back to login page?

**Reply by tht-activeviam** [reply]
We used the OIDC authentication to login. 
Where do Atoti store the JWT token? I can't find it in Local Storeage/Session Storage.

In the chrome developer tools, I saw the cookie JSESSIONID. I noticed the expiration of that cookie was set to "Session" which mean it will not expired until the browser is closed. 
Is that something Atoti should control to set the idle time?

![image](https://user-images.githubusercontent.com/41277323/159674932-e69a202c-87a3-4826-827f-814fa7a0e329.png)

**Reply by tibdex** [reply]
The UI fetches a JWT token from the server to know the name and roles of the user but it doesn't store it anywhere. The `JSESSIONID` cookie is the only thing authenticating the user against the server. Is there a problem with this cookie living until the browser is closed? Once #386 is done, users will be able to log out without closing their browser. Isn't this good enough?

**Reply by tht-activeviam** [reply]
Hi @tibdex ,
To pass the production test in client side, we are trying to address these:
- to set idle time out of 15 mins. At the moment, we don't expire the session until the browser closed.
- the JWT token from the issuer has expiration time, such as 12 hours. I think our Atoti should consider that for logging out the user. At the moment, I'm still testing it by waiting 12 hours to let the token expired.
